{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import itertools\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import statsmodels.api as sm\n",
    "import statsmodels.formula.api as smf\n",
    "import sys\n",
    "\n",
    "from itertools import combinations \n",
    "from scipy.stats import boxcox\n",
    "from scipy.stats import linregress\n",
    "from scipy.stats import spearmanr\n",
    "from scipy.stats import pearsonr\n",
    "from statsmodels.stats.anova import anova_lm\n",
    "\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.neighbors import NearestNeighbors\n",
    "\n",
    "# import utils\n",
    "sys.path.append(\"../../../utils\")\n",
    "from plotting_utils import *\n",
    "\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "mpl.rcParams['figure.autolayout'] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set(**PAPER_PRESET)\n",
    "fontsize = PAPER_FONTSIZE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.random.seed(2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "QUANT_ALPHA = 0.05"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_f = \"../../../data/04__mapped_motifs/human_motifs_filtered.txt.gz\"\n",
    "mouse_motifs_f = \"../../../data/04__mapped_motifs/mouse_motifs_filtered.txt.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "motif_info_dir = \"../../../misc/01__motif_info\"\n",
    "motif_map_f = \"%s/00__lambert_et_al_files/00__metadata/curated_motif_map.txt\" % motif_info_dir\n",
    "motif_info_f = \"%s/00__lambert_et_al_files/00__metadata/motif_info.txt\" % motif_info_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_motifs_f = \"../../../data/04__mapped_motifs/sig_motifs.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_f = \"../../../data/02__mpra/03__results/all_processed_results.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "orth_expr_f = \"../../../data/03__rna_seq/04__TF_expr/orth_TF_expression.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>elem_key</th>\n",
       "      <th>...</th>\n",
       "      <th>actual_end</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>species</th>\n",
       "      <th>tss_tile_num</th>\n",
       "      <th>hg19_id</th>\n",
       "      <th>biotype_hg19</th>\n",
       "      <th>minimal_biotype_hg19</th>\n",
       "      <th>stem_exp_hg19</th>\n",
       "      <th>hg19_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>-</td>\n",
       "      <td>20.0704</td>\n",
       "      <td>7.460000e-08</td>\n",
       "      <td>0.000824</td>\n",
       "      <td>GGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAG</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>218519582</td>\n",
       "      <td>726.1.0.0</td>\n",
       "      <td>h.793</td>\n",
       "      <td>HUMAN</td>\n",
       "      <td>tile1</td>\n",
       "      <td>h.793</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>h.793__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF571</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>105</td>\n",
       "      <td>119</td>\n",
       "      <td>+</td>\n",
       "      <td>17.8440</td>\n",
       "      <td>5.780000e-07</td>\n",
       "      <td>0.002630</td>\n",
       "      <td>CAGCCTCGGCGGCCC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>218519582</td>\n",
       "      <td>726.1.0.0</td>\n",
       "      <td>h.793</td>\n",
       "      <td>HUMAN</td>\n",
       "      <td>tile1</td>\n",
       "      <td>h.793</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>h.793__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>11</td>\n",
       "      <td>49</td>\n",
       "      <td>-</td>\n",
       "      <td>16.2746</td>\n",
       "      <td>1.190000e-06</td>\n",
       "      <td>0.004230</td>\n",
       "      <td>TGCGGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCA</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>218519582</td>\n",
       "      <td>726.1.0.0</td>\n",
       "      <td>h.793</td>\n",
       "      <td>HUMAN</td>\n",
       "      <td>tile1</td>\n",
       "      <td>h.793</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>h.793__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>5</td>\n",
       "      <td>43</td>\n",
       "      <td>-</td>\n",
       "      <td>15.2113</td>\n",
       "      <td>2.400000e-06</td>\n",
       "      <td>0.006280</td>\n",
       "      <td>GCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAG</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>218519582</td>\n",
       "      <td>726.1.0.0</td>\n",
       "      <td>h.793</td>\n",
       "      <td>HUMAN</td>\n",
       "      <td>tile1</td>\n",
       "      <td>h.793</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>h.793__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>26</td>\n",
       "      <td>46</td>\n",
       "      <td>-</td>\n",
       "      <td>15.1735</td>\n",
       "      <td>2.720000e-06</td>\n",
       "      <td>0.003270</td>\n",
       "      <td>GGCGCTGAGAGCAGGAGCAGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>218519582</td>\n",
       "      <td>726.1.0.0</td>\n",
       "      <td>h.793</td>\n",
       "      <td>HUMAN</td>\n",
       "      <td>tile1</td>\n",
       "      <td>h.793</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.0</td>\n",
       "      <td>h.793__tile1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name sequence name  start  stop strand_x    score       p-value  \\\n",
       "0         ZNF93    elem_12149      8    46        -  20.0704  7.460000e-08   \n",
       "1        ZNF571    elem_12149    105   119        +  17.8440  5.780000e-07   \n",
       "2         ZNF93    elem_12149     11    49        -  16.2746  1.190000e-06   \n",
       "3         ZNF93    elem_12149      5    43        -  15.2113  2.400000e-06   \n",
       "4      M10427_1    elem_12149     26    46        -  15.1735  2.720000e-06   \n",
       "\n",
       "    q-value                         matched sequence    elem_key  \\\n",
       "0  0.000824  GGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAG  elem_12149   \n",
       "1  0.002630                          CAGCCTCGGCGGCCC  elem_12149   \n",
       "2  0.004230  TGCGGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCA  elem_12149   \n",
       "3  0.006280  GCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAG  elem_12149   \n",
       "4  0.003270                    GGCGCTGAGAGCAGGAGCAGC  elem_12149   \n",
       "\n",
       "       ...      actual_end  dupe_info tss_id species tss_tile_num  hg19_id  \\\n",
       "0      ...       218519582  726.1.0.0  h.793   HUMAN        tile1    h.793   \n",
       "1      ...       218519582  726.1.0.0  h.793   HUMAN        tile1    h.793   \n",
       "2      ...       218519582  726.1.0.0  h.793   HUMAN        tile1    h.793   \n",
       "3      ...       218519582  726.1.0.0  h.793   HUMAN        tile1    h.793   \n",
       "4      ...       218519582  726.1.0.0  h.793   HUMAN        tile1    h.793   \n",
       "\n",
       "     biotype_hg19 minimal_biotype_hg19  stem_exp_hg19    hg19_index  \n",
       "0  protein_coding                 mRNA            0.0  h.793__tile1  \n",
       "1  protein_coding                 mRNA            0.0  h.793__tile1  \n",
       "2  protein_coding                 mRNA            0.0  h.793__tile1  \n",
       "3  protein_coding                 mRNA            0.0  h.793__tile1  \n",
       "4  protein_coding                 mRNA            0.0  h.793__tile1  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# this file is already filtered to correct tile nums\n",
    "human_motifs = pd.read_table(human_motifs_f, sep=\"\\t\")\n",
    "human_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>elem_key</th>\n",
       "      <th>...</th>\n",
       "      <th>actual_end</th>\n",
       "      <th>dupe_info</th>\n",
       "      <th>tss_id</th>\n",
       "      <th>species</th>\n",
       "      <th>tss_tile_num</th>\n",
       "      <th>mm9_id</th>\n",
       "      <th>biotype_mm9</th>\n",
       "      <th>minimal_biotype_mm9</th>\n",
       "      <th>stem_exp_mm9</th>\n",
       "      <th>mm9_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>4</td>\n",
       "      <td>32</td>\n",
       "      <td>-</td>\n",
       "      <td>31.7041</td>\n",
       "      <td>3.800000e-11</td>\n",
       "      <td>2.690000e-07</td>\n",
       "      <td>AGTCGCACACACACACACACACACACACA</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>120217220</td>\n",
       "      <td>9280.1.0.0</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>MOUSE</td>\n",
       "      <td>tile1</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.045</td>\n",
       "      <td>m.3178__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>-</td>\n",
       "      <td>19.9490</td>\n",
       "      <td>1.510000e-08</td>\n",
       "      <td>6.260000e-05</td>\n",
       "      <td>TCGCACACACACACACACACACACACATA</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>120217220</td>\n",
       "      <td>9280.1.0.0</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>MOUSE</td>\n",
       "      <td>tile1</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.045</td>\n",
       "      <td>m.3178__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>49</td>\n",
       "      <td>77</td>\n",
       "      <td>-</td>\n",
       "      <td>16.8265</td>\n",
       "      <td>5.010000e-08</td>\n",
       "      <td>1.890000e-04</td>\n",
       "      <td>ATACGCGCGCACACACATACATACACACA</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>120217220</td>\n",
       "      <td>9280.1.0.0</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>MOUSE</td>\n",
       "      <td>tile1</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.045</td>\n",
       "      <td>m.3178__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>47</td>\n",
       "      <td>75</td>\n",
       "      <td>-</td>\n",
       "      <td>13.4796</td>\n",
       "      <td>1.610000e-07</td>\n",
       "      <td>5.620000e-04</td>\n",
       "      <td>ACGCGCGCACACACATACATACACACATA</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>120217220</td>\n",
       "      <td>9280.1.0.0</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>MOUSE</td>\n",
       "      <td>tile1</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.045</td>\n",
       "      <td>m.3178__tile1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>41</td>\n",
       "      <td>69</td>\n",
       "      <td>-</td>\n",
       "      <td>12.0816</td>\n",
       "      <td>2.540000e-07</td>\n",
       "      <td>8.380000e-04</td>\n",
       "      <td>GCACACACATACATACACACATACACACA</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>120217220</td>\n",
       "      <td>9280.1.0.0</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>MOUSE</td>\n",
       "      <td>tile1</td>\n",
       "      <td>m.3178</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>0.045</td>\n",
       "      <td>m.3178__tile1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name sequence name  start  stop strand_x    score       p-value  \\\n",
       "0        ZNF785     elem_9927      4    32        -  31.7041  3.800000e-11   \n",
       "1        ZNF785     elem_9927      2    30        -  19.9490  1.510000e-08   \n",
       "2        ZNF785     elem_9927     49    77        -  16.8265  5.010000e-08   \n",
       "3        ZNF785     elem_9927     47    75        -  13.4796  1.610000e-07   \n",
       "4        ZNF785     elem_9927     41    69        -  12.0816  2.540000e-07   \n",
       "\n",
       "        q-value               matched sequence   elem_key      ...        \\\n",
       "0  2.690000e-07  AGTCGCACACACACACACACACACACACA  elem_9927      ...         \n",
       "1  6.260000e-05  TCGCACACACACACACACACACACACATA  elem_9927      ...         \n",
       "2  1.890000e-04  ATACGCGCGCACACACATACATACACACA  elem_9927      ...         \n",
       "3  5.620000e-04  ACGCGCGCACACACATACATACACACATA  elem_9927      ...         \n",
       "4  8.380000e-04  GCACACACATACATACACACATACACACA  elem_9927      ...         \n",
       "\n",
       "  actual_end   dupe_info  tss_id species tss_tile_num  mm9_id     biotype_mm9  \\\n",
       "0  120217220  9280.1.0.0  m.3178   MOUSE        tile1  m.3178  protein_coding   \n",
       "1  120217220  9280.1.0.0  m.3178   MOUSE        tile1  m.3178  protein_coding   \n",
       "2  120217220  9280.1.0.0  m.3178   MOUSE        tile1  m.3178  protein_coding   \n",
       "3  120217220  9280.1.0.0  m.3178   MOUSE        tile1  m.3178  protein_coding   \n",
       "4  120217220  9280.1.0.0  m.3178   MOUSE        tile1  m.3178  protein_coding   \n",
       "\n",
       "  minimal_biotype_mm9  stem_exp_mm9      mm9_index  \n",
       "0                mRNA         0.045  m.3178__tile1  \n",
       "1                mRNA         0.045  m.3178__tile1  \n",
       "2                mRNA         0.045  m.3178__tile1  \n",
       "3                mRNA         0.045  m.3178__tile1  \n",
       "4                mRNA         0.045  m.3178__tile1  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# this file is already filtered to correct tile nums\n",
    "mouse_motifs = pd.read_table(mouse_motifs_f, sep=\"\\t\")\n",
    "mouse_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl ID</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>Motif evidence</th>\n",
       "      <th>Motif ID</th>\n",
       "      <th>Motif type</th>\n",
       "      <th>Motif source</th>\n",
       "      <th>CIS-BP ID</th>\n",
       "      <th>Best Motif(s)? (Figure 2A)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000267281</td>\n",
       "      <td>AC023509.3</td>\n",
       "      <td>Inferred - ATF7 (100% AA Identity, Homo sapiens)</td>\n",
       "      <td>ATF7_eDBD_HT-SELEX</td>\n",
       "      <td>HT-SELEX</td>\n",
       "      <td>Yin2017</td>\n",
       "      <td>M08490_1.94d</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000267281</td>\n",
       "      <td>AC023509.3</td>\n",
       "      <td>Inferred - ATF7 (100% AA Identity, Homo sapiens)</td>\n",
       "      <td>ATF7_eDBD_Methyl-HT-SELEX</td>\n",
       "      <td>Methyl-HT-SELEX</td>\n",
       "      <td>Yin2017</td>\n",
       "      <td>M08491_1.94d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000267281</td>\n",
       "      <td>AC023509.3</td>\n",
       "      <td>Inferred - Atf7 (98% AA Identity, Mus musculus)</td>\n",
       "      <td>3T3L1-Atf7_GSE56872</td>\n",
       "      <td>Misc</td>\n",
       "      <td>HOMER</td>\n",
       "      <td>M06975_1.94d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000264668</td>\n",
       "      <td>AC138696.1</td>\n",
       "      <td>Inferred - ZFP41 (100% AA Identity, Homo sapiens)</td>\n",
       "      <td>ZFP41_FL_HT-SELEX</td>\n",
       "      <td>HT-SELEX</td>\n",
       "      <td>Yin2017</td>\n",
       "      <td>M09315_1.94d</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000264668</td>\n",
       "      <td>AC138696.1</td>\n",
       "      <td>Inferred - ZFP41 (100% AA Identity, Homo sapiens)</td>\n",
       "      <td>ZFP41_FL_Methyl-HT-SELEX</td>\n",
       "      <td>Methyl-HT-SELEX</td>\n",
       "      <td>Yin2017</td>\n",
       "      <td>M09316_1.94d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Ensembl ID HGNC symbol  \\\n",
       "0  ENSG00000267281  AC023509.3   \n",
       "1  ENSG00000267281  AC023509.3   \n",
       "2  ENSG00000267281  AC023509.3   \n",
       "3  ENSG00000264668  AC138696.1   \n",
       "4  ENSG00000264668  AC138696.1   \n",
       "\n",
       "                                      Motif evidence  \\\n",
       "0   Inferred - ATF7 (100% AA Identity, Homo sapiens)   \n",
       "1   Inferred - ATF7 (100% AA Identity, Homo sapiens)   \n",
       "2    Inferred - Atf7 (98% AA Identity, Mus musculus)   \n",
       "3  Inferred - ZFP41 (100% AA Identity, Homo sapiens)   \n",
       "4  Inferred - ZFP41 (100% AA Identity, Homo sapiens)   \n",
       "\n",
       "                    Motif ID       Motif type Motif source     CIS-BP ID  \\\n",
       "0         ATF7_eDBD_HT-SELEX         HT-SELEX      Yin2017  M08490_1.94d   \n",
       "1  ATF7_eDBD_Methyl-HT-SELEX  Methyl-HT-SELEX      Yin2017  M08491_1.94d   \n",
       "2        3T3L1-Atf7_GSE56872             Misc        HOMER  M06975_1.94d   \n",
       "3          ZFP41_FL_HT-SELEX         HT-SELEX      Yin2017  M09315_1.94d   \n",
       "4   ZFP41_FL_Methyl-HT-SELEX  Methyl-HT-SELEX      Yin2017  M09316_1.94d   \n",
       "\n",
       "  Best Motif(s)? (Figure 2A)  \n",
       "0                       True  \n",
       "1                        NaN  \n",
       "2                        NaN  \n",
       "3                       True  \n",
       "4                        NaN  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "motif_info = pd.read_table(motif_info_f, sep=\"\\t\")\n",
    "motif_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "466\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>lr_test</th>\n",
       "      <th>pval</th>\n",
       "      <th>rsq</th>\n",
       "      <th>beta</th>\n",
       "      <th>padj</th>\n",
       "      <th>short_id</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>activ_or_repr</th>\n",
       "      <th>no_CAGE_enr</th>\n",
       "      <th>eRNA_enr</th>\n",
       "      <th>lncRNA_enr</th>\n",
       "      <th>mRNA_enr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HKR1</td>\n",
       "      <td>6.281798</td>\n",
       "      <td>0.012198</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>0.048118</td>\n",
       "      <td>0.036709</td>\n",
       "      <td>HKR1</td>\n",
       "      <td>HKR1</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M00144_1</td>\n",
       "      <td>10.738762</td>\n",
       "      <td>0.001049</td>\n",
       "      <td>0.000657</td>\n",
       "      <td>-0.275327</td>\n",
       "      <td>0.004203</td>\n",
       "      <td>M00144_1</td>\n",
       "      <td>ARID5A</td>\n",
       "      <td>repressing</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M00224_1</td>\n",
       "      <td>8.999747</td>\n",
       "      <td>0.002700</td>\n",
       "      <td>0.000550</td>\n",
       "      <td>-0.228712</td>\n",
       "      <td>0.009926</td>\n",
       "      <td>M00224_1</td>\n",
       "      <td>TWIST2</td>\n",
       "      <td>repressing</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M00226_1</td>\n",
       "      <td>6.933266</td>\n",
       "      <td>0.008461</td>\n",
       "      <td>0.000424</td>\n",
       "      <td>0.133341</td>\n",
       "      <td>0.027122</td>\n",
       "      <td>M00226_1</td>\n",
       "      <td>ARNT</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M00242_1</td>\n",
       "      <td>18.650533</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.001140</td>\n",
       "      <td>-0.321524</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>M00242_1</td>\n",
       "      <td>SCX</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      index    lr_test      pval       rsq      beta      padj  short_id  \\\n",
       "0      HKR1   6.281798  0.012198  0.000384  0.048118  0.036709      HKR1   \n",
       "1  M00144_1  10.738762  0.001049  0.000657 -0.275327  0.004203  M00144_1   \n",
       "2  M00224_1   8.999747  0.002700  0.000550 -0.228712  0.009926  M00224_1   \n",
       "3  M00226_1   6.933266  0.008461  0.000424  0.133341  0.027122  M00226_1   \n",
       "4  M00242_1  18.650533  0.000016  0.001140 -0.321524  0.000094  M00242_1   \n",
       "\n",
       "  HGNC symbol activ_or_repr  no_CAGE_enr  eRNA_enr  lncRNA_enr  mRNA_enr  \n",
       "0        HKR1    activating          NaN       NaN         NaN       NaN  \n",
       "1      ARID5A    repressing          0.0       NaN         NaN       NaN  \n",
       "2      TWIST2    repressing          0.0       NaN         NaN       NaN  \n",
       "3        ARNT    activating          NaN       NaN         NaN       NaN  \n",
       "4         SCX    repressing          NaN       1.0         NaN       NaN  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_motifs = pd.read_table(sig_motifs_f)\n",
    "sig_motifs = sig_motifs[sig_motifs[\"padj\"] < 0.05]\n",
    "print(len(sig_motifs))\n",
    "sig_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "77\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>lr_test</th>\n",
       "      <th>pval</th>\n",
       "      <th>rsq</th>\n",
       "      <th>beta</th>\n",
       "      <th>padj</th>\n",
       "      <th>short_id</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>activ_or_repr</th>\n",
       "      <th>no_CAGE_enr</th>\n",
       "      <th>eRNA_enr</th>\n",
       "      <th>lncRNA_enr</th>\n",
       "      <th>mRNA_enr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>M00554_1</td>\n",
       "      <td>252.997930</td>\n",
       "      <td>5.766059e-57</td>\n",
       "      <td>0.015335</td>\n",
       "      <td>0.323914</td>\n",
       "      <td>1.381719e-55</td>\n",
       "      <td>M00554_1</td>\n",
       "      <td>KLF1</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>M00600_1</td>\n",
       "      <td>436.566496</td>\n",
       "      <td>6.049445e-97</td>\n",
       "      <td>0.026288</td>\n",
       "      <td>0.421643</td>\n",
       "      <td>2.302348e-95</td>\n",
       "      <td>M00600_1</td>\n",
       "      <td>KLF7</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>M00616_1</td>\n",
       "      <td>432.422883</td>\n",
       "      <td>4.825598e-96</td>\n",
       "      <td>0.026042</td>\n",
       "      <td>0.413977</td>\n",
       "      <td>1.784093e-94</td>\n",
       "      <td>M00616_1</td>\n",
       "      <td>KLF8</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>M00977_1</td>\n",
       "      <td>827.369873</td>\n",
       "      <td>6.046188e-182</td>\n",
       "      <td>0.049129</td>\n",
       "      <td>0.877262</td>\n",
       "      <td>2.607922e-179</td>\n",
       "      <td>M00977_1</td>\n",
       "      <td>ETV3L</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>M03040_1</td>\n",
       "      <td>587.149099</td>\n",
       "      <td>1.044774e-129</td>\n",
       "      <td>0.035165</td>\n",
       "      <td>0.766805</td>\n",
       "      <td>7.115462e-128</td>\n",
       "      <td>M03040_1</td>\n",
       "      <td>NFYB</td>\n",
       "      <td>activating</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       index     lr_test           pval       rsq      beta           padj  \\\n",
       "12  M00554_1  252.997930   5.766059e-57  0.015335  0.323914   1.381719e-55   \n",
       "14  M00600_1  436.566496   6.049445e-97  0.026288  0.421643   2.302348e-95   \n",
       "17  M00616_1  432.422883   4.825598e-96  0.026042  0.413977   1.784093e-94   \n",
       "21  M00977_1  827.369873  6.046188e-182  0.049129  0.877262  2.607922e-179   \n",
       "36  M03040_1  587.149099  1.044774e-129  0.035165  0.766805  7.115462e-128   \n",
       "\n",
       "    short_id HGNC symbol activ_or_repr  no_CAGE_enr  eRNA_enr  lncRNA_enr  \\\n",
       "12  M00554_1        KLF1    activating          NaN       NaN         2.0   \n",
       "14  M00600_1        KLF7    activating          NaN       NaN         2.0   \n",
       "17  M00616_1        KLF8    activating          NaN       NaN         2.0   \n",
       "21  M00977_1       ETV3L    activating          NaN       NaN         2.0   \n",
       "36  M03040_1        NFYB    activating          NaN       NaN         2.0   \n",
       "\n",
       "    mRNA_enr  \n",
       "12       3.0  \n",
       "14       3.0  \n",
       "17       3.0  \n",
       "21       3.0  \n",
       "36       3.0  "
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_motifs_str = sig_motifs[sig_motifs[\"rsq\"] >= 0.01]\n",
    "print(len(sig_motifs_str))\n",
    "sig_motifs_str.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hg19_id</th>\n",
       "      <th>chr_tss_hg19</th>\n",
       "      <th>start_tss_hg19</th>\n",
       "      <th>biotype_hg19</th>\n",
       "      <th>cage_id_hg19</th>\n",
       "      <th>name_peak_hg19</th>\n",
       "      <th>minimal_biotype_hg19</th>\n",
       "      <th>stem_exp_hg19</th>\n",
       "      <th>mm9_id</th>\n",
       "      <th>chr_tss_mm9</th>\n",
       "      <th>...</th>\n",
       "      <th>trans_mouse_status</th>\n",
       "      <th>trans_human_status_det</th>\n",
       "      <th>trans_mouse_status_det</th>\n",
       "      <th>trans_status_one</th>\n",
       "      <th>trans_status_det_one</th>\n",
       "      <th>logFC_trans_one</th>\n",
       "      <th>logFC_int</th>\n",
       "      <th>fdr_int</th>\n",
       "      <th>cis_trans_int_status</th>\n",
       "      <th>cis_trans_int_status_det</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>h.2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2985430.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr1:2985420..2985438,-</td>\n",
       "      <td>ENSG00000177133.6</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.07111110000000001</td>\n",
       "      <td>m.1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>-0.325920</td>\n",
       "      <td>-0.050780</td>\n",
       "      <td>0.991115</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>h.3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8086552.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr1:8086546..8086571,+</td>\n",
       "      <td>ENSG00000238290.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.10444400000000001</td>\n",
       "      <td>m.2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>0.533476</td>\n",
       "      <td>-0.405629</td>\n",
       "      <td>0.908445</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>h.4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>26498322.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>chr1:26498321..26498327,-</td>\n",
       "      <td>ENSG00000236782.1</td>\n",
       "      <td>other</td>\n",
       "      <td>0.03</td>\n",
       "      <td>m.3</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>0.270223</td>\n",
       "      <td>-0.210732</td>\n",
       "      <td>0.960813</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>h.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65533428.0</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>chr1:65533390..65533443,-</td>\n",
       "      <td>ENSG00000231485.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.841111</td>\n",
       "      <td>m.4</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.449668</td>\n",
       "      <td>-0.740913</td>\n",
       "      <td>0.152051</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>h.6</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65533462.0</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>chr1:65533457..65533465,-</td>\n",
       "      <td>ENSG00000231485.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.116667</td>\n",
       "      <td>m.4</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>-0.270646</td>\n",
       "      <td>-0.528464</td>\n",
       "      <td>0.379849</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  hg19_id chr_tss_hg19  start_tss_hg19 biotype_hg19  \\\n",
       "0     h.2         chr1       2985430.0    divergent   \n",
       "1     h.3         chr1       8086552.0    divergent   \n",
       "2     h.4         chr1      26498322.0    antisense   \n",
       "3     h.5         chr1      65533428.0   intergenic   \n",
       "4     h.6         chr1      65533462.0   intergenic   \n",
       "\n",
       "                cage_id_hg19     name_peak_hg19 minimal_biotype_hg19  \\\n",
       "0    chr1:2985420..2985438,-  ENSG00000177133.6               lncRNA   \n",
       "1    chr1:8086546..8086571,+  ENSG00000238290.1               lncRNA   \n",
       "2  chr1:26498321..26498327,-  ENSG00000236782.1                other   \n",
       "3  chr1:65533390..65533443,-  ENSG00000231485.1               lncRNA   \n",
       "4  chr1:65533457..65533465,-  ENSG00000231485.1               lncRNA   \n",
       "\n",
       "         stem_exp_hg19 mm9_id chr_tss_mm9            ...             \\\n",
       "0  0.07111110000000001    m.1        chr4            ...              \n",
       "1  0.10444400000000001    m.2        chr4            ...              \n",
       "2                 0.03    m.3        chr4            ...              \n",
       "3             0.841111    m.4        chr4            ...              \n",
       "4             0.116667    m.4        chr4            ...              \n",
       "\n",
       "   trans_mouse_status           trans_human_status_det trans_mouse_status_det  \\\n",
       "0     no trans effect                  no trans effect        no trans effect   \n",
       "1     no trans effect                  no trans effect        no trans effect   \n",
       "2     no trans effect                  no trans effect        no trans effect   \n",
       "3     no trans effect  trans effect\\n(higher in mouse)        no trans effect   \n",
       "4     no trans effect                  no trans effect        no trans effect   \n",
       "\n",
       "           trans_status_one             trans_status_det_one logFC_trans_one  \\\n",
       "0           no trans effect                  no trans effect       -0.325920   \n",
       "1           no trans effect                  no trans effect        0.533476   \n",
       "2           no trans effect                  no trans effect        0.270223   \n",
       "3  significant trans effect  trans effect\\n(higher in mouse)        0.449668   \n",
       "4           no trans effect                  no trans effect       -0.270646   \n",
       "\n",
       "  logFC_int   fdr_int      cis_trans_int_status  cis_trans_int_status_det  \n",
       "0 -0.050780  0.991115  no cis/trans int. effect  no cis/trans int. effect  \n",
       "1 -0.405629  0.908445  no cis/trans int. effect  no cis/trans int. effect  \n",
       "2 -0.210732  0.960813  no cis/trans int. effect  no cis/trans int. effect  \n",
       "3 -0.740913  0.152051  no cis/trans int. effect  no cis/trans int. effect  \n",
       "4 -0.528464  0.379849  no cis/trans int. effect  no cis/trans int. effect  \n",
       "\n",
       "[5 rows x 57 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = pd.read_table(data_f)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id_human</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>mean_tpm_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>mean_tpm_mouse</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>logFC</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>sig</th>\n",
       "      <th>neg_log_padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000142611</td>\n",
       "      <td>PRDM16</td>\n",
       "      <td>55.601719</td>\n",
       "      <td>ENSMUSG00000039410</td>\n",
       "      <td>Prdm16</td>\n",
       "      <td>612.953260</td>\n",
       "      <td>2.510447</td>\n",
       "      <td>4.964306</td>\n",
       "      <td>1.771139e-10</td>\n",
       "      <td>sig</td>\n",
       "      <td>9.751747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000078900</td>\n",
       "      <td>TP73</td>\n",
       "      <td>117.897387</td>\n",
       "      <td>ENSMUSG00000029026</td>\n",
       "      <td>Trp73</td>\n",
       "      <td>239.674040</td>\n",
       "      <td>2.530112</td>\n",
       "      <td>2.552952</td>\n",
       "      <td>4.216300e-07</td>\n",
       "      <td>sig</td>\n",
       "      <td>6.375069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000173673</td>\n",
       "      <td>HES3</td>\n",
       "      <td>272.208698</td>\n",
       "      <td>ENSMUSG00000028946</td>\n",
       "      <td>Hes3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.119951</td>\n",
       "      <td>-7.611337</td>\n",
       "      <td>1.590728e-10</td>\n",
       "      <td>sig</td>\n",
       "      <td>9.798404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000204859</td>\n",
       "      <td>ZBTB48</td>\n",
       "      <td>1251.437325</td>\n",
       "      <td>ENSMUSG00000028952</td>\n",
       "      <td>Zbtb48</td>\n",
       "      <td>342.804146</td>\n",
       "      <td>4.430457</td>\n",
       "      <td>-0.387987</td>\n",
       "      <td>5.686419e-02</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.245161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000125945</td>\n",
       "      <td>ZNF436</td>\n",
       "      <td>136.434616</td>\n",
       "      <td>ENSMUSG00000051351</td>\n",
       "      <td>Zfp46</td>\n",
       "      <td>177.665949</td>\n",
       "      <td>2.222969</td>\n",
       "      <td>1.836888</td>\n",
       "      <td>4.656334e-06</td>\n",
       "      <td>sig</td>\n",
       "      <td>5.331956</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_id_human gene_name_human  mean_tpm_human       gene_id_mouse  \\\n",
       "0  ENSG00000142611          PRDM16       55.601719  ENSMUSG00000039410   \n",
       "1  ENSG00000078900            TP73      117.897387  ENSMUSG00000029026   \n",
       "2  ENSG00000173673            HES3      272.208698  ENSMUSG00000028946   \n",
       "3  ENSG00000204859          ZBTB48     1251.437325  ENSMUSG00000028952   \n",
       "4  ENSG00000125945          ZNF436      136.434616  ENSMUSG00000051351   \n",
       "\n",
       "  gene_name_mouse  mean_tpm_mouse   AveExpr     logFC     adj.P.Val      sig  \\\n",
       "0          Prdm16      612.953260  2.510447  4.964306  1.771139e-10      sig   \n",
       "1           Trp73      239.674040  2.530112  2.552952  4.216300e-07      sig   \n",
       "2            Hes3        0.000000 -1.119951 -7.611337  1.590728e-10      sig   \n",
       "3          Zbtb48      342.804146  4.430457 -0.387987  5.686419e-02  not sig   \n",
       "4           Zfp46      177.665949  2.222969  1.836888  4.656334e-06      sig   \n",
       "\n",
       "   neg_log_padj  \n",
       "0      9.751747  \n",
       "1      6.375069  \n",
       "2      9.798404  \n",
       "3      1.245161  \n",
       "4      5.331956  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orth_expr = pd.read_table(orth_expr_f)\n",
    "orth_expr.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. motif files already filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "448"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(human_motifs[\"#pattern name\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "448"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(mouse_motifs[\"#pattern name\"].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. plot motifs for one example"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>elem_key</th>\n",
       "      <th>...</th>\n",
       "      <th>rsq</th>\n",
       "      <th>beta</th>\n",
       "      <th>padj</th>\n",
       "      <th>short_id</th>\n",
       "      <th>HGNC symbol</th>\n",
       "      <th>activ_or_repr</th>\n",
       "      <th>no_CAGE_enr</th>\n",
       "      <th>eRNA_enr</th>\n",
       "      <th>lncRNA_enr</th>\n",
       "      <th>mRNA_enr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>-</td>\n",
       "      <td>20.07040</td>\n",
       "      <td>7.460000e-08</td>\n",
       "      <td>0.000824</td>\n",
       "      <td>GGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAG</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>-0.083051</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>11</td>\n",
       "      <td>49</td>\n",
       "      <td>-</td>\n",
       "      <td>16.27460</td>\n",
       "      <td>1.190000e-06</td>\n",
       "      <td>0.004230</td>\n",
       "      <td>TGCGGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCA</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>-0.083051</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>5</td>\n",
       "      <td>43</td>\n",
       "      <td>-</td>\n",
       "      <td>15.21130</td>\n",
       "      <td>2.400000e-06</td>\n",
       "      <td>0.006280</td>\n",
       "      <td>GCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAG</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>-0.083051</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "      <td>-</td>\n",
       "      <td>9.14789</td>\n",
       "      <td>8.130000e-05</td>\n",
       "      <td>0.030900</td>\n",
       "      <td>GAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAGCGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>-0.083051</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF93</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>13</td>\n",
       "      <td>51</td>\n",
       "      <td>+</td>\n",
       "      <td>9.04225</td>\n",
       "      <td>8.580000e-05</td>\n",
       "      <td>0.031400</td>\n",
       "      <td>CTCCGGAGCTGCTGCTGCTCCTGCTCTCAGCGCCGCAGT</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000969</td>\n",
       "      <td>-0.083051</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>ZNF93</td>\n",
       "      <td>repressing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name sequence name  start  stop strand_x     score       p-value  \\\n",
       "0         ZNF93    elem_12149      8    46        -  20.07040  7.460000e-08   \n",
       "1         ZNF93    elem_12149     11    49        -  16.27460  1.190000e-06   \n",
       "2         ZNF93    elem_12149      5    43        -  15.21130  2.400000e-06   \n",
       "3         ZNF93    elem_12149      2    40        -   9.14789  8.130000e-05   \n",
       "4         ZNF93    elem_12149     13    51        +   9.04225  8.580000e-05   \n",
       "\n",
       "    q-value                         matched sequence    elem_key    ...     \\\n",
       "0  0.000824  GGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAG  elem_12149    ...      \n",
       "1  0.004230  TGCGGCGCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCA  elem_12149    ...      \n",
       "2  0.006280  GCTGAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAG  elem_12149    ...      \n",
       "3  0.030900  GAGAGCAGGAGCAGCAGCAGCTCCGGAGCAGAGAAGCGC  elem_12149    ...      \n",
       "4  0.031400  CTCCGGAGCTGCTGCTGCTCCTGCTCTCAGCGCCGCAGT  elem_12149    ...      \n",
       "\n",
       "        rsq      beta     padj short_id HGNC symbol  activ_or_repr  \\\n",
       "0  0.000969 -0.083051  0.00037    ZNF93       ZNF93     repressing   \n",
       "1  0.000969 -0.083051  0.00037    ZNF93       ZNF93     repressing   \n",
       "2  0.000969 -0.083051  0.00037    ZNF93       ZNF93     repressing   \n",
       "3  0.000969 -0.083051  0.00037    ZNF93       ZNF93     repressing   \n",
       "4  0.000969 -0.083051  0.00037    ZNF93       ZNF93     repressing   \n",
       "\n",
       "  no_CAGE_enr eRNA_enr  lncRNA_enr  mRNA_enr  \n",
       "0         NaN      NaN         NaN       3.0  \n",
       "1         NaN      NaN         NaN       3.0  \n",
       "2         NaN      NaN         NaN       3.0  \n",
       "3         NaN      NaN         NaN       3.0  \n",
       "4         NaN      NaN         NaN       3.0  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_motifs = human_motifs.merge(sig_motifs, left_on=\"#pattern name\", right_on=\"index\")\n",
    "mouse_motifs = mouse_motifs.merge(sig_motifs, left_on=\"#pattern name\", right_on=\"index\")\n",
    "human_motifs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#pattern name', 'sequence name', 'start', 'stop', 'strand_x', 'score',\n",
       "       'p-value', 'q-value', 'matched sequence', 'elem_key', 'elem', 'element',\n",
       "       'tile_type', 'element_id', 'name', 'tile_number', 'chrom', 'strand_y',\n",
       "       'actual_start', 'actual_end', 'dupe_info', 'tss_id', 'species',\n",
       "       'tss_tile_num', 'hg19_id', 'biotype_hg19', 'minimal_biotype_hg19',\n",
       "       'stem_exp_hg19', 'hg19_index', 'index', 'lr_test', 'pval', 'rsq',\n",
       "       'beta', 'padj', 'short_id', 'HGNC symbol', 'activ_or_repr',\n",
       "       'no_CAGE_enr', 'eRNA_enr', 'lncRNA_enr', 'mRNA_enr'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_motifs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_grp = human_motifs.groupby([\"hg19_id\", \"minimal_biotype_hg19\"])[\"index\"].agg(\"count\").reset_index()\n",
    "enh_grp = human_motifs_grp[human_motifs_grp[\"minimal_biotype_hg19\"] == \"eRNA\"]\n",
    "enh_grp.to_csv(\"enh_grp.tmp.txt\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "167\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hg19_id</th>\n",
       "      <th>chr_tss_hg19</th>\n",
       "      <th>start_tss_hg19</th>\n",
       "      <th>biotype_hg19</th>\n",
       "      <th>cage_id_hg19</th>\n",
       "      <th>name_peak_hg19</th>\n",
       "      <th>minimal_biotype_hg19</th>\n",
       "      <th>stem_exp_hg19</th>\n",
       "      <th>mm9_id</th>\n",
       "      <th>chr_tss_mm9</th>\n",
       "      <th>...</th>\n",
       "      <th>trans_mouse_status</th>\n",
       "      <th>trans_human_status_det</th>\n",
       "      <th>trans_mouse_status_det</th>\n",
       "      <th>trans_status_one</th>\n",
       "      <th>trans_status_det_one</th>\n",
       "      <th>logFC_trans_one</th>\n",
       "      <th>logFC_int</th>\n",
       "      <th>fdr_int</th>\n",
       "      <th>cis_trans_int_status</th>\n",
       "      <th>cis_trans_int_status_det</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>h.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>65533428.0</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>chr1:65533390..65533443,-</td>\n",
       "      <td>ENSG00000231485.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.841111</td>\n",
       "      <td>m.4</td>\n",
       "      <td>chr4</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.449668</td>\n",
       "      <td>-0.740913</td>\n",
       "      <td>0.152051</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>h.60</td>\n",
       "      <td>chr11</td>\n",
       "      <td>82783147.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr11:82783097..82783155,+</td>\n",
       "      <td>ENSG00000246067.3</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>14.0844</td>\n",
       "      <td>m.38</td>\n",
       "      <td>chr7</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>-0.986585</td>\n",
       "      <td>0.804973</td>\n",
       "      <td>0.153595</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>h.61</td>\n",
       "      <td>chr11</td>\n",
       "      <td>82904704.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr11:82904668..82904715,-</td>\n",
       "      <td>ENSG00000247137.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>5.318890000000001</td>\n",
       "      <td>m.39</td>\n",
       "      <td>chr7</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.987324</td>\n",
       "      <td>-0.223281</td>\n",
       "      <td>0.940418</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>h.79</td>\n",
       "      <td>chr12</td>\n",
       "      <td>93964604.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>chr12:93964602..93964622,-</td>\n",
       "      <td>ENSG00000246985.3</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.138889</td>\n",
       "      <td>m.49</td>\n",
       "      <td>chr10</td>\n",
       "      <td>...</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.621065</td>\n",
       "      <td>-0.254516</td>\n",
       "      <td>0.905592</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>h.119</td>\n",
       "      <td>chr15</td>\n",
       "      <td>83419323.0</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>chr15:83419302..83419341,+</td>\n",
       "      <td>ENSG00000250988.3</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>2.24556</td>\n",
       "      <td>m.75</td>\n",
       "      <td>chr7</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>1.028888</td>\n",
       "      <td>-0.327597</td>\n",
       "      <td>0.683751</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   hg19_id chr_tss_hg19  start_tss_hg19 biotype_hg19  \\\n",
       "3      h.5         chr1      65533428.0   intergenic   \n",
       "37    h.60        chr11      82783147.0    divergent   \n",
       "38    h.61        chr11      82904704.0    divergent   \n",
       "51    h.79        chr12      93964604.0    antisense   \n",
       "73   h.119        chr15      83419323.0   intergenic   \n",
       "\n",
       "                  cage_id_hg19     name_peak_hg19 minimal_biotype_hg19  \\\n",
       "3    chr1:65533390..65533443,-  ENSG00000231485.1               lncRNA   \n",
       "37  chr11:82783097..82783155,+  ENSG00000246067.3               lncRNA   \n",
       "38  chr11:82904668..82904715,-  ENSG00000247137.4               lncRNA   \n",
       "51  chr12:93964602..93964622,-  ENSG00000246985.3               lncRNA   \n",
       "73  chr15:83419302..83419341,+  ENSG00000250988.3               lncRNA   \n",
       "\n",
       "        stem_exp_hg19 mm9_id chr_tss_mm9            ...             \\\n",
       "3            0.841111    m.4        chr4            ...              \n",
       "37            14.0844   m.38        chr7            ...              \n",
       "38  5.318890000000001   m.39        chr7            ...              \n",
       "51           0.138889   m.49       chr10            ...              \n",
       "73            2.24556   m.75        chr7            ...              \n",
       "\n",
       "          trans_mouse_status           trans_human_status_det  \\\n",
       "3            no trans effect  trans effect\\n(higher in mouse)   \n",
       "37           no trans effect  trans effect\\n(higher in human)   \n",
       "38  significant trans effect  trans effect\\n(higher in mouse)   \n",
       "51           no trans effect  trans effect\\n(higher in mouse)   \n",
       "73  significant trans effect  trans effect\\n(higher in mouse)   \n",
       "\n",
       "             trans_mouse_status_det          trans_status_one  \\\n",
       "3                   no trans effect  significant trans effect   \n",
       "37                  no trans effect  significant trans effect   \n",
       "38  trans effect\\n(higher in mouse)  significant trans effect   \n",
       "51                  no trans effect  significant trans effect   \n",
       "73  trans effect\\n(higher in mouse)  significant trans effect   \n",
       "\n",
       "               trans_status_det_one logFC_trans_one logFC_int   fdr_int  \\\n",
       "3   trans effect\\n(higher in mouse)        0.449668 -0.740913  0.152051   \n",
       "37  trans effect\\n(higher in human)       -0.986585  0.804973  0.153595   \n",
       "38  trans effect\\n(higher in mouse)        0.987324 -0.223281  0.940418   \n",
       "51  trans effect\\n(higher in mouse)        0.621065 -0.254516  0.905592   \n",
       "73  trans effect\\n(higher in mouse)        1.028888 -0.327597  0.683751   \n",
       "\n",
       "        cis_trans_int_status  cis_trans_int_status_det  \n",
       "3   no cis/trans int. effect  no cis/trans int. effect  \n",
       "37  no cis/trans int. effect  no cis/trans int. effect  \n",
       "38  no cis/trans int. effect  no cis/trans int. effect  \n",
       "51  no cis/trans int. effect  no cis/trans int. effect  \n",
       "73  no cis/trans int. effect  no cis/trans int. effect  \n",
       "\n",
       "[5 rows x 57 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_filt = data[~pd.isnull(data[\"minimal_biotype_hg19\"])]\n",
    "trans_filt = trans_filt[((trans_filt[\"HUES64_padj_hg19\"] < QUANT_ALPHA) | (trans_filt[\"mESC_padj_mm9\"] < QUANT_ALPHA))]\n",
    "trans_filt = trans_filt[trans_filt[\"trans_human_status\"] == \"significant trans effect\"]\n",
    "print(len(trans_filt))\n",
    "trans_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hg19_id</th>\n",
       "      <th>minimal_biotype_hg19</th>\n",
       "      <th>index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1445</th>\n",
       "      <td>h.2556</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1894</th>\n",
       "      <td>h.3062</td>\n",
       "      <td>other</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1691</th>\n",
       "      <td>h.2826</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2594</th>\n",
       "      <td>h.505</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     hg19_id minimal_biotype_hg19  index\n",
       "1445  h.2556                 eRNA     17\n",
       "1894  h.3062                other     43\n",
       "1691  h.2826               lncRNA     53\n",
       "2594   h.505                 mRNA     96"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = human_motifs[human_motifs[\"HGNC symbol\"] == \"DMBX1\"]\n",
    "tmp_grp = human_motifs_grp[human_motifs_grp[\"hg19_id\"].isin(tmp[\"hg19_id\"])]\n",
    "tmp_grp = tmp_grp[tmp_grp[\"hg19_id\"].isin(trans_filt[\"hg19_id\"])]\n",
    "tmp_grp.sort_values(by=\"index\").head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "235979"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(human_motifs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "def genome_start(row):\n",
    "    if row.strand_y == \"+\":\n",
    "        genome_start = row.actual_start + row.start - 1\n",
    "    else:\n",
    "        genome_start = row.actual_end - row.stop\n",
    "    return genome_start\n",
    "        \n",
    "def genome_end(row):\n",
    "    if row.strand_y == \"+\":\n",
    "        genome_end = row.actual_start + row.stop\n",
    "    else:\n",
    "        genome_end = row.actual_end - row.start + 1\n",
    "    return genome_end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs[\"genome_start\"] = human_motifs.apply(genome_start, axis=1)\n",
    "human_motifs[\"genome_end\"] = human_motifs.apply(genome_end, axis=1)\n",
    "mouse_motifs[\"genome_start\"] = mouse_motifs.apply(genome_start, axis=1)\n",
    "mouse_motifs[\"genome_end\"] = mouse_motifs.apply(genome_end, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_bed = human_motifs[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n",
    "mouse_motifs_bed = mouse_motifs[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "# human_motifs_bed.to_csv(\"human_motifs.bed\", sep=\"\\t\", header=False, index=False)\n",
    "# mouse_motifs_bed.to_csv(\"mouse_motifs.bed\", sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "139479\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>elem_key</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>mean_tpm_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>mean_tpm_mouse</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>logFC</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>sig</th>\n",
       "      <th>neg_log_padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>26</td>\n",
       "      <td>46</td>\n",
       "      <td>-</td>\n",
       "      <td>15.1735</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.00327</td>\n",
       "      <td>GGCGCTGAGAGCAGGAGCAGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF341</td>\n",
       "      <td>540.302268</td>\n",
       "      <td>ENSMUSG00000059842</td>\n",
       "      <td>Zfp341</td>\n",
       "      <td>130.51635</td>\n",
       "      <td>3.169967</td>\n",
       "      <td>-0.557345</td>\n",
       "      <td>0.026036</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.584424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>20</td>\n",
       "      <td>40</td>\n",
       "      <td>-</td>\n",
       "      <td>15.0306</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.00341</td>\n",
       "      <td>GAGAGCAGGAGCAGCAGCAGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF341</td>\n",
       "      <td>540.302268</td>\n",
       "      <td>ENSMUSG00000059842</td>\n",
       "      <td>Zfp341</td>\n",
       "      <td>130.51635</td>\n",
       "      <td>3.169967</td>\n",
       "      <td>-0.557345</td>\n",
       "      <td>0.026036</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.584424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>113</td>\n",
       "      <td>133</td>\n",
       "      <td>-</td>\n",
       "      <td>14.0612</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.00465</td>\n",
       "      <td>GAGCGCGGTGAGGGGGGCCGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF341</td>\n",
       "      <td>540.302268</td>\n",
       "      <td>ENSMUSG00000059842</td>\n",
       "      <td>Zfp341</td>\n",
       "      <td>130.51635</td>\n",
       "      <td>3.169967</td>\n",
       "      <td>-0.557345</td>\n",
       "      <td>0.026036</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.584424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>23</td>\n",
       "      <td>43</td>\n",
       "      <td>-</td>\n",
       "      <td>12.4592</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.00758</td>\n",
       "      <td>GCTGAGAGCAGGAGCAGCAGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF341</td>\n",
       "      <td>540.302268</td>\n",
       "      <td>ENSMUSG00000059842</td>\n",
       "      <td>Zfp341</td>\n",
       "      <td>130.51635</td>\n",
       "      <td>3.169967</td>\n",
       "      <td>-0.557345</td>\n",
       "      <td>0.026036</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.584424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>M10427_1</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>116</td>\n",
       "      <td>136</td>\n",
       "      <td>-</td>\n",
       "      <td>11.5408</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>0.00992</td>\n",
       "      <td>CGGGAGCGCGGTGAGGGGGGC</td>\n",
       "      <td>elem_12149</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF341</td>\n",
       "      <td>540.302268</td>\n",
       "      <td>ENSMUSG00000059842</td>\n",
       "      <td>Zfp341</td>\n",
       "      <td>130.51635</td>\n",
       "      <td>3.169967</td>\n",
       "      <td>-0.557345</td>\n",
       "      <td>0.026036</td>\n",
       "      <td>not sig</td>\n",
       "      <td>1.584424</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name sequence name  start  stop strand_x    score   p-value  \\\n",
       "0      M10427_1    elem_12149     26    46        -  15.1735  0.000003   \n",
       "1      M10427_1    elem_12149     20    40        -  15.0306  0.000003   \n",
       "2      M10427_1    elem_12149    113   133        -  14.0612  0.000007   \n",
       "3      M10427_1    elem_12149     23    43        -  12.4592  0.000022   \n",
       "4      M10427_1    elem_12149    116   136        -  11.5408  0.000041   \n",
       "\n",
       "   q-value       matched sequence    elem_key      ...      gene_name_human  \\\n",
       "0  0.00327  GGCGCTGAGAGCAGGAGCAGC  elem_12149      ...               ZNF341   \n",
       "1  0.00341  GAGAGCAGGAGCAGCAGCAGC  elem_12149      ...               ZNF341   \n",
       "2  0.00465  GAGCGCGGTGAGGGGGGCCGC  elem_12149      ...               ZNF341   \n",
       "3  0.00758  GCTGAGAGCAGGAGCAGCAGC  elem_12149      ...               ZNF341   \n",
       "4  0.00992  CGGGAGCGCGGTGAGGGGGGC  elem_12149      ...               ZNF341   \n",
       "\n",
       "  mean_tpm_human       gene_id_mouse gene_name_mouse mean_tpm_mouse   AveExpr  \\\n",
       "0     540.302268  ENSMUSG00000059842          Zfp341      130.51635  3.169967   \n",
       "1     540.302268  ENSMUSG00000059842          Zfp341      130.51635  3.169967   \n",
       "2     540.302268  ENSMUSG00000059842          Zfp341      130.51635  3.169967   \n",
       "3     540.302268  ENSMUSG00000059842          Zfp341      130.51635  3.169967   \n",
       "4     540.302268  ENSMUSG00000059842          Zfp341      130.51635  3.169967   \n",
       "\n",
       "      logFC adj.P.Val      sig  neg_log_padj  \n",
       "0 -0.557345  0.026036  not sig      1.584424  \n",
       "1 -0.557345  0.026036  not sig      1.584424  \n",
       "2 -0.557345  0.026036  not sig      1.584424  \n",
       "3 -0.557345  0.026036  not sig      1.584424  \n",
       "4 -0.557345  0.026036  not sig      1.584424  \n",
       "\n",
       "[5 rows x 55 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_motifs_orth = human_motifs.merge(orth_expr, left_on=\"HGNC symbol\", right_on=\"gene_name_human\")\n",
    "print(len(human_motifs_orth))\n",
    "human_motifs_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133575\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#pattern name</th>\n",
       "      <th>sequence name</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>score</th>\n",
       "      <th>p-value</th>\n",
       "      <th>q-value</th>\n",
       "      <th>matched sequence</th>\n",
       "      <th>elem_key</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_name_human</th>\n",
       "      <th>mean_tpm_human</th>\n",
       "      <th>gene_id_mouse</th>\n",
       "      <th>gene_name_mouse</th>\n",
       "      <th>mean_tpm_mouse</th>\n",
       "      <th>AveExpr</th>\n",
       "      <th>logFC</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>sig</th>\n",
       "      <th>neg_log_padj</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF202</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>103</td>\n",
       "      <td>110</td>\n",
       "      <td>-</td>\n",
       "      <td>14.85710</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.0545</td>\n",
       "      <td>CCGCCTCC</td>\n",
       "      <td>elem_9927</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF202</td>\n",
       "      <td>3234.571511</td>\n",
       "      <td>ENSMUSG00000025602</td>\n",
       "      <td>Zfp202</td>\n",
       "      <td>142.520611</td>\n",
       "      <td>4.644599</td>\n",
       "      <td>-3.054947</td>\n",
       "      <td>1.294671e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>8.887841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ZNF202</td>\n",
       "      <td>elem_9509</td>\n",
       "      <td>72</td>\n",
       "      <td>79</td>\n",
       "      <td>+</td>\n",
       "      <td>14.57140</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.0545</td>\n",
       "      <td>CCTCCTCC</td>\n",
       "      <td>elem_9509</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF202</td>\n",
       "      <td>3234.571511</td>\n",
       "      <td>ENSMUSG00000025602</td>\n",
       "      <td>Zfp202</td>\n",
       "      <td>142.520611</td>\n",
       "      <td>4.644599</td>\n",
       "      <td>-3.054947</td>\n",
       "      <td>1.294671e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>8.887841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF202</td>\n",
       "      <td>elem_9509</td>\n",
       "      <td>69</td>\n",
       "      <td>76</td>\n",
       "      <td>+</td>\n",
       "      <td>6.04082</td>\n",
       "      <td>0.000082</td>\n",
       "      <td>0.0725</td>\n",
       "      <td>CCCCCTCC</td>\n",
       "      <td>elem_9509</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF202</td>\n",
       "      <td>3234.571511</td>\n",
       "      <td>ENSMUSG00000025602</td>\n",
       "      <td>Zfp202</td>\n",
       "      <td>142.520611</td>\n",
       "      <td>4.644599</td>\n",
       "      <td>-3.054947</td>\n",
       "      <td>1.294671e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>8.887841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ZNF202</td>\n",
       "      <td>elem_1635</td>\n",
       "      <td>26</td>\n",
       "      <td>33</td>\n",
       "      <td>+</td>\n",
       "      <td>6.04082</td>\n",
       "      <td>0.000082</td>\n",
       "      <td>0.0725</td>\n",
       "      <td>CCCCCTCC</td>\n",
       "      <td>elem_1635</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF202</td>\n",
       "      <td>3234.571511</td>\n",
       "      <td>ENSMUSG00000025602</td>\n",
       "      <td>Zfp202</td>\n",
       "      <td>142.520611</td>\n",
       "      <td>4.644599</td>\n",
       "      <td>-3.054947</td>\n",
       "      <td>1.294671e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>8.887841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF202</td>\n",
       "      <td>elem_3159</td>\n",
       "      <td>91</td>\n",
       "      <td>98</td>\n",
       "      <td>+</td>\n",
       "      <td>6.04082</td>\n",
       "      <td>0.000082</td>\n",
       "      <td>0.0725</td>\n",
       "      <td>CCCCCTCC</td>\n",
       "      <td>elem_3159</td>\n",
       "      <td>...</td>\n",
       "      <td>ZNF202</td>\n",
       "      <td>3234.571511</td>\n",
       "      <td>ENSMUSG00000025602</td>\n",
       "      <td>Zfp202</td>\n",
       "      <td>142.520611</td>\n",
       "      <td>4.644599</td>\n",
       "      <td>-3.054947</td>\n",
       "      <td>1.294671e-09</td>\n",
       "      <td>sig</td>\n",
       "      <td>8.887841</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #pattern name sequence name  start  stop strand_x     score   p-value  \\\n",
       "0        ZNF202     elem_9927    103   110        -  14.85710  0.000016   \n",
       "1        ZNF202     elem_9509     72    79        +  14.57140  0.000036   \n",
       "2        ZNF202     elem_9509     69    76        +   6.04082  0.000082   \n",
       "3        ZNF202     elem_1635     26    33        +   6.04082  0.000082   \n",
       "4        ZNF202     elem_3159     91    98        +   6.04082  0.000082   \n",
       "\n",
       "   q-value matched sequence   elem_key      ...      gene_name_human  \\\n",
       "0   0.0545         CCGCCTCC  elem_9927      ...               ZNF202   \n",
       "1   0.0545         CCTCCTCC  elem_9509      ...               ZNF202   \n",
       "2   0.0725         CCCCCTCC  elem_9509      ...               ZNF202   \n",
       "3   0.0725         CCCCCTCC  elem_1635      ...               ZNF202   \n",
       "4   0.0725         CCCCCTCC  elem_3159      ...               ZNF202   \n",
       "\n",
       "  mean_tpm_human       gene_id_mouse gene_name_mouse mean_tpm_mouse   AveExpr  \\\n",
       "0    3234.571511  ENSMUSG00000025602          Zfp202     142.520611  4.644599   \n",
       "1    3234.571511  ENSMUSG00000025602          Zfp202     142.520611  4.644599   \n",
       "2    3234.571511  ENSMUSG00000025602          Zfp202     142.520611  4.644599   \n",
       "3    3234.571511  ENSMUSG00000025602          Zfp202     142.520611  4.644599   \n",
       "4    3234.571511  ENSMUSG00000025602          Zfp202     142.520611  4.644599   \n",
       "\n",
       "      logFC     adj.P.Val  sig  neg_log_padj  \n",
       "0 -3.054947  1.294671e-09  sig      8.887841  \n",
       "1 -3.054947  1.294671e-09  sig      8.887841  \n",
       "2 -3.054947  1.294671e-09  sig      8.887841  \n",
       "3 -3.054947  1.294671e-09  sig      8.887841  \n",
       "4 -3.054947  1.294671e-09  sig      8.887841  \n",
       "\n",
       "[5 rows x 55 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_motifs_orth = mouse_motifs.merge(orth_expr, left_on=\"HGNC symbol\", right_on=\"gene_name_human\")\n",
    "print(len(mouse_motifs_orth))\n",
    "mouse_motifs_orth.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_DE = human_motifs_orth[human_motifs_orth[\"sig\"] == \"sig\"]\n",
    "human_motifs_not_DE = human_motifs_orth[human_motifs_orth[\"sig\"] != \"sig\"]\n",
    "mouse_motifs_DE = mouse_motifs_orth[mouse_motifs_orth[\"sig\"] == \"sig\"]\n",
    "mouse_motifs_not_DE = mouse_motifs_orth[mouse_motifs_orth[\"sig\"] != \"sig\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "167\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hg19_id</th>\n",
       "      <th>chr_tss_hg19</th>\n",
       "      <th>start_tss_hg19</th>\n",
       "      <th>biotype_hg19</th>\n",
       "      <th>cage_id_hg19</th>\n",
       "      <th>name_peak_hg19</th>\n",
       "      <th>minimal_biotype_hg19</th>\n",
       "      <th>stem_exp_hg19</th>\n",
       "      <th>mm9_id</th>\n",
       "      <th>chr_tss_mm9</th>\n",
       "      <th>...</th>\n",
       "      <th>trans_mouse_status</th>\n",
       "      <th>trans_human_status_det</th>\n",
       "      <th>trans_mouse_status_det</th>\n",
       "      <th>trans_status_one</th>\n",
       "      <th>trans_status_det_one</th>\n",
       "      <th>logFC_trans_one</th>\n",
       "      <th>logFC_int</th>\n",
       "      <th>fdr_int</th>\n",
       "      <th>cis_trans_int_status</th>\n",
       "      <th>cis_trans_int_status_det</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>h.31</td>\n",
       "      <td>chr1</td>\n",
       "      <td>210407227.0</td>\n",
       "      <td>antisense</td>\n",
       "      <td>chr1:210407207..210407250,-</td>\n",
       "      <td>ENSG00000203706.4</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>0.946667</td>\n",
       "      <td>m.23</td>\n",
       "      <td>chr1</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>-1.337778</td>\n",
       "      <td>-1.228367</td>\n",
       "      <td>0.144818</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>h.33</td>\n",
       "      <td>chr10</td>\n",
       "      <td>18947991.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr10:18947984..18947997,-</td>\n",
       "      <td>ENSG00000152487.6</td>\n",
       "      <td>other</td>\n",
       "      <td>0.0177778</td>\n",
       "      <td>m.25</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.577630</td>\n",
       "      <td>0.147640</td>\n",
       "      <td>0.969522</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>h.34</td>\n",
       "      <td>chr10</td>\n",
       "      <td>18948008.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr10:18948004..18948012,-</td>\n",
       "      <td>ENSG00000152487.6</td>\n",
       "      <td>other</td>\n",
       "      <td>0.0</td>\n",
       "      <td>m.25</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.576520</td>\n",
       "      <td>0.195427</td>\n",
       "      <td>0.931772</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>h.35</td>\n",
       "      <td>chr10</td>\n",
       "      <td>18948165.0</td>\n",
       "      <td>divergent</td>\n",
       "      <td>chr10:18948156..18948202,-</td>\n",
       "      <td>ENSG00000152487.6</td>\n",
       "      <td>other</td>\n",
       "      <td>1.0333299999999999</td>\n",
       "      <td>m.25</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in mouse)</td>\n",
       "      <td>0.557331</td>\n",
       "      <td>0.147817</td>\n",
       "      <td>0.975331</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>h.38</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126605462.0</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>chr10:126605451..126605474,+</td>\n",
       "      <td>ENSG00000249456.1</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>4.35444</td>\n",
       "      <td>m.27</td>\n",
       "      <td>chr7</td>\n",
       "      <td>...</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>no trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>significant trans effect</td>\n",
       "      <td>trans effect\\n(higher in human)</td>\n",
       "      <td>-0.969088</td>\n",
       "      <td>-1.263427</td>\n",
       "      <td>0.095455</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "      <td>no cis/trans int. effect</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   hg19_id chr_tss_hg19  start_tss_hg19 biotype_hg19  \\\n",
       "19    h.31         chr1     210407227.0    antisense   \n",
       "20    h.33        chr10      18947991.0    divergent   \n",
       "21    h.34        chr10      18948008.0    divergent   \n",
       "22    h.35        chr10      18948165.0    divergent   \n",
       "24    h.38        chr10     126605462.0   intergenic   \n",
       "\n",
       "                    cage_id_hg19     name_peak_hg19 minimal_biotype_hg19  \\\n",
       "19   chr1:210407207..210407250,-  ENSG00000203706.4               lncRNA   \n",
       "20    chr10:18947984..18947997,-  ENSG00000152487.6                other   \n",
       "21    chr10:18948004..18948012,-  ENSG00000152487.6                other   \n",
       "22    chr10:18948156..18948202,-  ENSG00000152487.6                other   \n",
       "24  chr10:126605451..126605474,+  ENSG00000249456.1               lncRNA   \n",
       "\n",
       "         stem_exp_hg19 mm9_id chr_tss_mm9            ...             \\\n",
       "19            0.946667   m.23        chr1            ...              \n",
       "20           0.0177778   m.25        chr2            ...              \n",
       "21                 0.0   m.25        chr2            ...              \n",
       "22  1.0333299999999999   m.25        chr2            ...              \n",
       "24             4.35444   m.27        chr7            ...              \n",
       "\n",
       "          trans_mouse_status trans_human_status_det  \\\n",
       "19  significant trans effect        no trans effect   \n",
       "20  significant trans effect        no trans effect   \n",
       "21  significant trans effect        no trans effect   \n",
       "22  significant trans effect        no trans effect   \n",
       "24  significant trans effect        no trans effect   \n",
       "\n",
       "             trans_mouse_status_det          trans_status_one  \\\n",
       "19  trans effect\\n(higher in human)  significant trans effect   \n",
       "20  trans effect\\n(higher in mouse)  significant trans effect   \n",
       "21  trans effect\\n(higher in mouse)  significant trans effect   \n",
       "22  trans effect\\n(higher in mouse)  significant trans effect   \n",
       "24  trans effect\\n(higher in human)  significant trans effect   \n",
       "\n",
       "               trans_status_det_one logFC_trans_one logFC_int   fdr_int  \\\n",
       "19  trans effect\\n(higher in human)       -1.337778 -1.228367  0.144818   \n",
       "20  trans effect\\n(higher in mouse)        0.577630  0.147640  0.969522   \n",
       "21  trans effect\\n(higher in mouse)        0.576520  0.195427  0.931772   \n",
       "22  trans effect\\n(higher in mouse)        0.557331  0.147817  0.975331   \n",
       "24  trans effect\\n(higher in human)       -0.969088 -1.263427  0.095455   \n",
       "\n",
       "        cis_trans_int_status  cis_trans_int_status_det  \n",
       "19  no cis/trans int. effect  no cis/trans int. effect  \n",
       "20  no cis/trans int. effect  no cis/trans int. effect  \n",
       "21  no cis/trans int. effect  no cis/trans int. effect  \n",
       "22  no cis/trans int. effect  no cis/trans int. effect  \n",
       "24  no cis/trans int. effect  no cis/trans int. effect  \n",
       "\n",
       "[5 rows x 57 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_filt = data[~pd.isnull(data[\"minimal_biotype_hg19\"])]\n",
    "trans_filt = trans_filt[((trans_filt[\"HUES64_padj_hg19\"] < QUANT_ALPHA) | (trans_filt[\"mESC_padj_mm9\"] < QUANT_ALPHA))]\n",
    "trans_filt = trans_filt[trans_filt[\"trans_mouse_status\"] == \"significant trans effect\"]\n",
    "print(len(trans_filt))\n",
    "trans_filt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mm9_id</th>\n",
       "      <th>minimal_biotype_mm9</th>\n",
       "      <th>score_DE</th>\n",
       "      <th>score_not</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1767</th>\n",
       "      <td>m.2911</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2234</th>\n",
       "      <td>m.3382</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2229</th>\n",
       "      <td>m.3378</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2130</th>\n",
       "      <td>m.3282</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1505</th>\n",
       "      <td>m.2623</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>m.2150</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2364</th>\n",
       "      <td>m.394</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>m.1292</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1353</th>\n",
       "      <td>m.2451</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>5.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2674</th>\n",
       "      <td>m.755</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2401</th>\n",
       "      <td>m.440</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>6.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2469</th>\n",
       "      <td>m.52</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>6.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1489</th>\n",
       "      <td>m.2600</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>6.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1935</th>\n",
       "      <td>m.3074</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>7.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2145</th>\n",
       "      <td>m.3296</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2750</th>\n",
       "      <td>m.838</td>\n",
       "      <td>other</td>\n",
       "      <td>8.0</td>\n",
       "      <td>22.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1574</th>\n",
       "      <td>m.27</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2222</th>\n",
       "      <td>m.3370</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2433</th>\n",
       "      <td>m.479</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1683</th>\n",
       "      <td>m.2824</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1331</th>\n",
       "      <td>m.2427</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2218</th>\n",
       "      <td>m.3367</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2464</th>\n",
       "      <td>m.514</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1362</th>\n",
       "      <td>m.2461</td>\n",
       "      <td>other</td>\n",
       "      <td>10.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>m.2545</td>\n",
       "      <td>other</td>\n",
       "      <td>10.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1568</th>\n",
       "      <td>m.2692</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>10.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>m.1310</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>10.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>860</th>\n",
       "      <td>m.1930</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>10.0</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>m.1224</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>11.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>m.1139</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>11.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1491</th>\n",
       "      <td>m.2602</td>\n",
       "      <td>other</td>\n",
       "      <td>46.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1504</th>\n",
       "      <td>m.2618</td>\n",
       "      <td>other</td>\n",
       "      <td>46.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1507</th>\n",
       "      <td>m.2626</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>46.0</td>\n",
       "      <td>44.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>959</th>\n",
       "      <td>m.204</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>46.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>m.1632</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>49.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1545</th>\n",
       "      <td>m.2667</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>49.0</td>\n",
       "      <td>23.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614</th>\n",
       "      <td>m.1656</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>50.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2631</th>\n",
       "      <td>m.708</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>50.0</td>\n",
       "      <td>36.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2616</th>\n",
       "      <td>m.691</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>51.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>756</th>\n",
       "      <td>m.1815</td>\n",
       "      <td>no CAGE activity</td>\n",
       "      <td>53.0</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>m.3142</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>54.0</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2683</th>\n",
       "      <td>m.766</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>54.0</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1633</th>\n",
       "      <td>m.2767</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>57.0</td>\n",
       "      <td>33.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>m.1382</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>58.0</td>\n",
       "      <td>45.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>m.1597</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>59.0</td>\n",
       "      <td>68.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>m.1070</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>59.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>633</th>\n",
       "      <td>m.1677</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>59.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1647</th>\n",
       "      <td>m.2782</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>61.0</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>m.1004</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>62.0</td>\n",
       "      <td>43.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1401</th>\n",
       "      <td>m.25</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>62.0</td>\n",
       "      <td>41.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2360</th>\n",
       "      <td>m.39</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>62.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2714</th>\n",
       "      <td>m.800</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>64.0</td>\n",
       "      <td>42.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>m.1030</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>64.0</td>\n",
       "      <td>60.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2311</th>\n",
       "      <td>m.346</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>65.0</td>\n",
       "      <td>46.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2154</th>\n",
       "      <td>m.3303</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>65.0</td>\n",
       "      <td>50.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2688</th>\n",
       "      <td>m.772</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>68.0</td>\n",
       "      <td>49.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>m.1152</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>69.0</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210</th>\n",
       "      <td>m.336</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>77.0</td>\n",
       "      <td>53.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1261</th>\n",
       "      <td>m.2355</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>82.0</td>\n",
       "      <td>67.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>m.1261</td>\n",
       "      <td>eRNA</td>\n",
       "      <td>86.0</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>159 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      mm9_id minimal_biotype_mm9  score_DE  score_not\n",
       "1767  m.2911              lncRNA       1.0        3.0\n",
       "2234  m.3382                eRNA       2.0        4.0\n",
       "2229  m.3378                eRNA       2.0        NaN\n",
       "2130  m.3282                eRNA       2.0        1.0\n",
       "1505  m.2623              lncRNA       3.0        4.0\n",
       "1058  m.2150                mRNA       4.0        3.0\n",
       "2364   m.394                mRNA       4.0        6.0\n",
       "290   m.1292                eRNA       5.0        6.0\n",
       "1353  m.2451              lncRNA       5.0        9.0\n",
       "2674   m.755                mRNA       5.0        6.0\n",
       "2401   m.440                mRNA       6.0        9.0\n",
       "2469    m.52              lncRNA       6.0       11.0\n",
       "1489  m.2600              lncRNA       6.0        9.0\n",
       "1935  m.3074                mRNA       7.0        5.0\n",
       "2145  m.3296                eRNA       8.0       11.0\n",
       "2750   m.838               other       8.0       22.0\n",
       "1574    m.27                mRNA       8.0        7.0\n",
       "2222  m.3370                eRNA       9.0        9.0\n",
       "2433   m.479                mRNA       9.0        9.0\n",
       "1683  m.2824                mRNA       9.0       12.0\n",
       "1331  m.2427              lncRNA       9.0        9.0\n",
       "2218  m.3367                eRNA       9.0        5.0\n",
       "2464   m.514                mRNA       9.0       11.0\n",
       "1362  m.2461               other      10.0       14.0\n",
       "1442  m.2545               other      10.0       11.0\n",
       "1568  m.2692              lncRNA      10.0        7.0\n",
       "306   m.1310                eRNA      10.0       13.0\n",
       "860   m.1930              lncRNA      10.0       12.0\n",
       "227   m.1224                eRNA      11.0       10.0\n",
       "145   m.1139                eRNA      11.0       10.0\n",
       "...      ...                 ...       ...        ...\n",
       "1491  m.2602               other      46.0       37.0\n",
       "1504  m.2618               other      46.0       45.0\n",
       "1507  m.2626              lncRNA      46.0       44.0\n",
       "959    m.204                mRNA      46.0       36.0\n",
       "593   m.1632              lncRNA      49.0       30.0\n",
       "1545  m.2667              lncRNA      49.0       23.0\n",
       "614   m.1656              lncRNA      50.0       42.0\n",
       "2631   m.708                mRNA      50.0       36.0\n",
       "2616   m.691                mRNA      51.0       37.0\n",
       "756   m.1815    no CAGE activity      53.0       54.0\n",
       "2002  m.3142                mRNA      54.0       35.0\n",
       "2683   m.766                mRNA      54.0       41.0\n",
       "1633  m.2767                mRNA      57.0       33.0\n",
       "377   m.1382              lncRNA      58.0       45.0\n",
       "565   m.1597              lncRNA      59.0       68.0\n",
       "75    m.1070                eRNA      59.0       31.0\n",
       "633   m.1677              lncRNA      59.0       42.0\n",
       "1647  m.2782                mRNA      61.0       39.0\n",
       "7     m.1004                mRNA      62.0       43.0\n",
       "1401    m.25                mRNA      62.0       41.0\n",
       "2360    m.39              lncRNA      62.0       40.0\n",
       "2714   m.800                mRNA      64.0       42.0\n",
       "35    m.1030                eRNA      64.0       60.0\n",
       "2311   m.346                eRNA      65.0       46.0\n",
       "2154  m.3303                eRNA      65.0       50.0\n",
       "2688   m.772                mRNA      68.0       49.0\n",
       "157   m.1152                eRNA      69.0       54.0\n",
       "2210   m.336                eRNA      77.0       53.0\n",
       "1261  m.2355                eRNA      82.0       67.0\n",
       "259   m.1261                eRNA      86.0       40.0\n",
       "\n",
       "[159 rows x 4 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_motifs_DE_grp = mouse_motifs_DE.groupby([\"mm9_id\", \"minimal_biotype_mm9\"])[\"score\"].agg(\"count\").reset_index()\n",
    "mouse_motifs_not_DE_grp = mouse_motifs_not_DE.groupby([\"mm9_id\", \"minimal_biotype_mm9\"])[\"score\"].agg(\"count\").reset_index()\n",
    "\n",
    "mouse_motifs_grp = mouse_motifs_DE_grp.merge(mouse_motifs_not_DE_grp, on=[\"mm9_id\", \"minimal_biotype_mm9\"], how=\"outer\",\n",
    "                                             suffixes=(\"_DE\", \"_not\"))\n",
    "mouse_motifs_grp_filt = mouse_motifs_grp[mouse_motifs_grp[\"mm9_id\"].isin(trans_filt[\"mm9_id\"])]\n",
    "mouse_motifs_grp_filt.sort_values(by=\"score_DE\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_DE_bed = human_motifs_DE[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n",
    "human_motifs_not_DE_bed = human_motifs_not_DE[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n",
    "\n",
    "mouse_motifs_DE_bed = mouse_motifs_DE[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n",
    "mouse_motifs_not_DE_bed = mouse_motifs_not_DE[[\"chrom\", \"genome_start\", \"genome_end\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_motifs_DE_bed.to_csv(\"human_motifs.DE.bed\", sep=\"\\t\", header=False, index=False)\n",
    "human_motifs_not_DE_bed.to_csv(\"human_motifs.notDE.bed\", sep=\"\\t\", header=False, index=False)\n",
    "mouse_motifs_DE_bed.to_csv(\"mouse_motifs.DE.bed\", sep=\"\\t\", header=False, index=False)\n",
    "mouse_motifs_not_DE_bed.to_csv(\"mouse_motifs.notDE.bed\", sep=\"\\t\", header=False, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "def getOverlap(a, b):\n",
    "    return max(a[0], b[0]) - min(a[1], b[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pybedtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ELF1', 'ELF5', 'ELF3', 'EHF', 'ELF2', 'ZNF200', 'PATZ1'], dtype=object)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hu_test = human_motifs[human_motifs[\"hg19_id\"] == \"h.986\"].sort_values(by=\"start\").drop_duplicates()\n",
    "hu_test_motifs = hu_test[\"HGNC symbol\"].unique()\n",
    "hu_test_motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ELF5', 'ELF2']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_cis = [\"ASCL2\", \"ASCL5\", \"ASCL4\", \"ZNF528\", \"ZNF263\", \"ZNF398\", \"ZNF132\", \"ZNF468\", \"ELF5\", \"ELF2\", \"ELF4\",\n",
    "           \"ERF\", \"ETV1\", \"ETV3L\", \"ETV6\", \"ETV5\", \"MAZ\"]\n",
    "[x for x in hu_test_motifs if x in sig_cis]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "tot_bed = pd.DataFrame()\n",
    "for motif in hu_test_motifs:\n",
    "    sub = hu_test[hu_test[\"HGNC symbol\"] == motif]\n",
    "    bed = sub[[\"chrom\", \"start\", \"stop\", \"HGNC symbol\", \"score\", \"strand_x\"]]\n",
    "    bed.columns = [\"chrom\", \"start\", \"end\", \"name\", \"score\", \"strand\"]\n",
    "    if len(bed) <= 1:\n",
    "        bed = bed[[\"chrom\", \"start\", \"end\", \"name\"]]\n",
    "        tot_bed = tot_bed.append(bed)\n",
    "    else:\n",
    "        bed.to_csv(\"tmp.bed\", sep=\"\\t\", header=False, index=False)\n",
    "    b = pybedtools.BedTool(\"tmp.bed\")\n",
    "    m = b.merge()\n",
    "    bed = m.to_dataframe()\n",
    "    bed[\"name\"] = motif\n",
    "    tot_bed = tot_bed.append(bed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42805</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>ELF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43877</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>ELF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47962</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>ELF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>EHF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>ELF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>ZNF200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>8</td>\n",
       "      <td>19</td>\n",
       "      <td>PATZ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40608</th>\n",
       "      <td>chr19</td>\n",
       "      <td>9</td>\n",
       "      <td>20</td>\n",
       "      <td>ZNF200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23553</th>\n",
       "      <td>chr19</td>\n",
       "      <td>15</td>\n",
       "      <td>22</td>\n",
       "      <td>PATZ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>27</td>\n",
       "      <td>54</td>\n",
       "      <td>ELF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>27</td>\n",
       "      <td>54</td>\n",
       "      <td>ELF5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>27</td>\n",
       "      <td>54</td>\n",
       "      <td>ELF3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>88</td>\n",
       "      <td>99</td>\n",
       "      <td>ELF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>88</td>\n",
       "      <td>99</td>\n",
       "      <td>ZNF200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>88</td>\n",
       "      <td>99</td>\n",
       "      <td>PATZ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>89</td>\n",
       "      <td>98</td>\n",
       "      <td>EHF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom  start  end    name\n",
       "42805  chr19      8   19    ELF1\n",
       "43877  chr19      8   18    ELF5\n",
       "47962  chr19      8   19    ELF3\n",
       "0      chr19      8   19     EHF\n",
       "0      chr19      8   19    ELF2\n",
       "0      chr19      8   19  ZNF200\n",
       "0      chr19      8   19   PATZ1\n",
       "40608  chr19      9   20  ZNF200\n",
       "23553  chr19     15   22   PATZ1\n",
       "0      chr19     27   54    ELF1\n",
       "0      chr19     27   54    ELF5\n",
       "0      chr19     27   54    ELF3\n",
       "1      chr19     88   99    ELF2\n",
       "1      chr19     88   99  ZNF200\n",
       "1      chr19     88   99   PATZ1\n",
       "1      chr19     89   98     EHF"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_bed = tot_bed.sort_values(by=\"start\")\n",
    "human_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['PATZ1'], dtype=object)"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mo_test = mouse_motifs[mouse_motifs[\"mm9_id\"] == \"m.869\"]\n",
    "mo_test_motifs = mo_test[\"HGNC symbol\"].unique()\n",
    "mo_test_motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "hu_motif_pos = zip(list(human_bed[\"start\"]), list(human_bed[\"end\"]))\n",
    "mo_motif_pos = zip(list(mo_test[\"start\"]), list(mo_test[\"stop\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"utf-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       "  \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Created with matplotlib (http://matplotlib.org/) -->\n",
       "<svg height=\"238pt\" version=\"1.1\" viewBox=\"0 0 411 238\" width=\"411pt\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       " <defs>\n",
       "  <style type=\"text/css\">\n",
       "*{stroke-linecap:butt;stroke-linejoin:round;}\n",
       "  </style>\n",
       " </defs>\n",
       " <g id=\"figure_1\">\n",
       "  <g id=\"patch_1\">\n",
       "   <path d=\"M 0 238.24 \n",
       "L 411.4 238.24 \n",
       "L 411.4 0 \n",
       "L 0 0 \n",
       "z\n",
       "\" style=\"fill:#ffffff;\"/>\n",
       "  </g>\n",
       "  <g id=\"axes_1\">\n",
       "   <g id=\"line2d_1\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 20.283636 \n",
       "L 63.110727 20.283636 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_2\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 30.167273 \n",
       "L 60.407612 30.167273 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_3\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 40.050909 \n",
       "L 63.110727 40.050909 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_4\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 49.934545 \n",
       "L 63.110727 49.934545 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_5\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 59.818182 \n",
       "L 63.110727 59.818182 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_6\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 69.701818 \n",
       "L 63.110727 69.701818 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_7\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 33.376471 79.585455 \n",
       "L 63.110727 79.585455 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_8\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 36.079585 89.469091 \n",
       "L 65.813841 89.469091 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_9\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 52.29827 99.352727 \n",
       "L 71.220069 99.352727 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_10\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 84.73564 20.283636 \n",
       "L 157.719723 20.283636 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_11\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 84.73564 30.167273 \n",
       "L 157.719723 30.167273 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_12\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 84.73564 40.050909 \n",
       "L 157.719723 40.050909 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_13\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 249.625606 20.283636 \n",
       "L 279.359862 20.283636 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_14\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 249.625606 30.167273 \n",
       "L 279.359862 30.167273 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_15\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 249.625606 40.050909 \n",
       "L 279.359862 40.050909 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "   <g id=\"line2d_16\">\n",
       "    <path clip-path=\"url(#pa6c2028646)\" d=\"M 252.32872 49.934545 \n",
       "L 276.656747 49.934545 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "  </g>\n",
       "  <g id=\"axes_2\">\n",
       "   <g id=\"line2d_17\">\n",
       "    <path clip-path=\"url(#pb1c2e89231)\" d=\"M 109.063668 178.421818 \n",
       "L 127.985467 178.421818 \n",
       "\" style=\"fill:none;stroke:#a9a9a9;stroke-width:5;\"/>\n",
       "   </g>\n",
       "  </g>\n",
       " </g>\n",
       " <defs>\n",
       "  <clipPath id=\"pa6c2028646\">\n",
       "   <rect height=\"98.836364\" width=\"390.6\" x=\"10.4\" y=\"10.4\"/>\n",
       "  </clipPath>\n",
       "  <clipPath id=\"pb1c2e89231\">\n",
       "   <rect height=\"98.836364\" width=\"390.6\" x=\"10.4\" y=\"129.003636\"/>\n",
       "  </clipPath>\n",
       " </defs>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ac927ed51d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fig, axarr = plt.subplots(figsize=(7, 4), nrows=2, ncols=1)\n",
    "\n",
    "# human ax\n",
    "ax = axarr[0]\n",
    "\n",
    "# plot motif locations\n",
    "xs = list(range(0, 144))\n",
    "prev_plotted = {}\n",
    "\n",
    "# iterate through things plotted at each prev_y value\n",
    "# if any overlaps, move\n",
    "linewidth=5\n",
    "for i, pos in enumerate(hu_motif_pos):\n",
    "    #print(\"\")\n",
    "    #print(\"i: %s, pos: %s\" % (i, pos))\n",
    "    plotted = False\n",
    "    if i == 0:\n",
    "        #print(\"first motif, plotting at y=0\")\n",
    "        ax.plot([pos[0], pos[1]], [0, 0], color=\"darkgrey\", linewidth=linewidth, solid_capstyle=\"butt\")\n",
    "        plotted = True\n",
    "        prev_plotted[0] = [pos]\n",
    "        continue\n",
    "    for prev_y in sorted(prev_plotted.keys(), reverse=True):\n",
    "        vals = prev_plotted[prev_y]\n",
    "        overlaps = []\n",
    "        for prev_pos in vals:\n",
    "            overlaps.append(getOverlap(prev_pos, pos))\n",
    "        if any(x < 0 for x in overlaps):\n",
    "            #print(\"motif overlaps w/ %s, continuing\" % (prev_y))\n",
    "            continue\n",
    "        else:\n",
    "            if not plotted:\n",
    "                #print(\"motif doesn't overlap anything at y=%s, plotting\" % prev_y)\n",
    "                ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=linewidth, \n",
    "                              solid_capstyle=\"butt\")\n",
    "                if prev_y not in prev_plotted:\n",
    "                    prev_plotted[prev_y] = [pos]\n",
    "                else:\n",
    "                    new_vals = list(prev_plotted[prev_y])\n",
    "                    new_vals.extend([pos])\n",
    "                    prev_plotted[prev_y] = new_vals\n",
    "                plotted = True\n",
    "    if not plotted:\n",
    "        prev_y -= 0.25\n",
    "        #print(\"motif overlaps at all prev_y, plotting at %s\" % prev_y)\n",
    "        ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=linewidth, \n",
    "                      solid_capstyle=\"butt\")\n",
    "        if prev_y not in prev_plotted:\n",
    "            prev_plotted[prev_y] = [pos]\n",
    "        else:\n",
    "            new_vals = list(prev_plotted[prev_y])\n",
    "            new_vals.extend([pos])\n",
    "            prev_plotted[prev_y] = new_vals\n",
    "        plotted = True\n",
    "#     print(prev_plotted)\n",
    "\n",
    "min_y = np.min(list(prev_plotted.keys()))\n",
    "\n",
    "# labels\n",
    "ax.set_xlim((-0.5, 144))\n",
    "ax.set_ylim((min_y - 0.25, 0.25))\n",
    "ax.set_xlabel(\"nucleotide number\")\n",
    "ax.set_ylabel(\"\")\n",
    "ax.xaxis.set_visible(False)\n",
    "ax.yaxis.set_visible(False)\n",
    "ax.axis(\"off\")\n",
    "\n",
    "\n",
    "\n",
    "# mouse ax\n",
    "ax = axarr[1]\n",
    "\n",
    "# plot motif locations\n",
    "xs = list(range(0, 144))\n",
    "prev_plotted = {}\n",
    "\n",
    "# iterate through things plotted at each prev_y value\n",
    "# if any overlaps, move\n",
    "linewidth=5\n",
    "for i, pos in enumerate(mo_motif_pos):\n",
    "    #print(\"\")\n",
    "    #print(\"i: %s, pos: %s\" % (i, pos))\n",
    "    plotted = False\n",
    "    if i == 0:\n",
    "        #print(\"first motif, plotting at y=0\")\n",
    "        ax.plot([pos[0], pos[1]], [0, 0], color=\"darkgrey\", linewidth=linewidth, solid_capstyle=\"butt\")\n",
    "        plotted = True\n",
    "        prev_plotted[0] = [pos]\n",
    "        continue\n",
    "    for prev_y in sorted(prev_plotted.keys(), reverse=True):\n",
    "        vals = prev_plotted[prev_y]\n",
    "        overlaps = []\n",
    "        for prev_pos in vals:\n",
    "            overlaps.append(getOverlap(prev_pos, pos))\n",
    "        if any(x < 0 for x in overlaps):\n",
    "            #print(\"motif overlaps w/ %s, continuing\" % (prev_y))\n",
    "            continue\n",
    "        else:\n",
    "            if not plotted:\n",
    "                #print(\"motif doesn't overlap anything at y=%s, plotting\" % prev_y)\n",
    "                ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=linewidth, \n",
    "                              solid_capstyle=\"butt\")\n",
    "                if prev_y not in prev_plotted:\n",
    "                    prev_plotted[prev_y] = [pos]\n",
    "                else:\n",
    "                    new_vals = list(prev_plotted[prev_y])\n",
    "                    new_vals.extend([pos])\n",
    "                    prev_plotted[prev_y] = new_vals\n",
    "                plotted = True\n",
    "    if not plotted:\n",
    "        prev_y -= 0.25\n",
    "        #print(\"motif overlaps at all prev_y, plotting at %s\" % prev_y)\n",
    "        ax.plot([pos[0], pos[1]], [prev_y, prev_y], color=\"darkgrey\", linewidth=linewidth, \n",
    "                      solid_capstyle=\"butt\")\n",
    "        if prev_y not in prev_plotted:\n",
    "            prev_plotted[prev_y] = [pos]\n",
    "        else:\n",
    "            new_vals = list(prev_plotted[prev_y])\n",
    "            new_vals.extend([pos])\n",
    "            prev_plotted[prev_y] = new_vals\n",
    "        plotted = True\n",
    "#     print(prev_plotted)\n",
    "\n",
    "min_y = np.min(list(prev_plotted.keys()))\n",
    "\n",
    "# labels\n",
    "ax.set_xlim((-0.5, 144))\n",
    "ax.set_ylim((min_y - 0.25, 0.25))\n",
    "ax.set_xlabel(\"nucleotide number\")\n",
    "ax.set_ylabel(\"\")\n",
    "ax.xaxis.set_visible(False)\n",
    "ax.yaxis.set_visible(False)\n",
    "ax.axis(\"off\")\n",
    "    \n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prev_plotted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:py3]",
   "language": "python",
   "name": "conda-env-py3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
